include/alpaka/idx/MapIdx.hpp

```diff
@@ -12,237 +12,82 @@
-        template<std::size_t TidxDimOut>
-        struct MapIdxPitchBytes<TidxDimOut, 1u, std::enable_if_t<(TidxDimOut > 1u)>>
+        if constexpr(TDimOut == 0 || TDimIn == 0)
+            return Vec<DimInt<TDimOut>, TElem>::zeros();
+        else if constexpr(TDimOut == TDimIn)
+            return in;
+        else if constexpr(TDimOut == 1)
         {
-            // \tparam TElem Type of the index values.
-            // \param idx Idx to be mapped.
-            // \param pitch Spatial pitch (in elems) to map the index to
-            // \return N dimensional vector.
-            ALPAKA_NO_HOST_ACC_WARNING
-            template<typename TElem>
-            ALPAKA_FN_HOST_ACC static auto mapIdxPitchBytes(
-                Vec<DimInt<1u>, TElem> const& idx,
-                Vec<DimInt<TidxDimOut>, TElem> const& pitch) -> Vec<DimInt<TidxDimOut>, TElem>
-            {
-                auto idxNd = Vec<DimInt<TidxDimOut>, TElem>::zeros();
-
-                constexpr std::size_t lastIdx = TidxDimOut - 1u;
-
-                TElem tmp = idx[0u];
-                for(std::size_t d(0u); d < lastIdx; ++d)
-                {
-                    idxNd[d] = static_cast<TElem>(tmp / pitch[d]);
-                    tmp %= pitch[d];
-                }
-                idxNd[lastIdx] = tmp;
-
-                return idxNd;
-            }
-        };
-        //! Maps a N dimensional index to a 1 dimensional index assuming a buffer without padding.
-        template<std::size_t TidxDimIn>
-        struct MapIdxPitchBytes<1u, TidxDimIn, std::enable_if_t<(TidxDimIn > 1u)>>
+            TElem out = in[0];
+            for(std::size_t d = 1; d < TDimIn; ++d)
+                out = static_cast<TElem>(out * extent[d] + in[d]);
+            return {out};
+        }
+        else if constexpr(TDimIn == 1)
         {
-            // \tparam TElem Type of the index values.
-            // \param idx Idx to be mapped.
-            // \param pitch Spatial pitch (in elems) to map the index to
-            // \return A 1 dimensional vector.
-            ALPAKA_NO_HOST_ACC_WARNING
-            template<typename TElem>
-            ALPAKA_FN_HOST_ACC static auto mapIdxPitchBytes(
-                Vec<DimInt<TidxDimIn>, TElem> const& idx,
-                Vec<DimInt<TidxDimIn>, TElem> const& pitch) -> Vec<DimInt<1u>, TElem>
+            auto flat = in.front();
+            Vec<DimInt<TDimOut>, TElem> out;
+            for(std::size_t d = TDimOut - 1u; d > 0; d--)
             {
-                using DimMinusOne = DimInt<TidxDimIn - 1>;
-                return {idx.back() + (subVecBegin<DimMinusOne>(pitch) * subVecBegin<DimMinusOne>(idx)).sum()};
+                out[d] = static_cast<TElem>(flat % extent[d]);
+                flat /= extent[d];
             }
-        };
+            out.front() = static_cast<TElem>(flat);
+            return out;
+        }
+        else
+            static_assert(sizeof(TElem) == 0, "Not implemented");
```

Comment:
Why not

```suggestion
            static_assert(!sizeof(TElem), "Not implemented");
```

like we do elsewhere?

Reply:
Because the first version requires less cycles to parse in my brain. But I don't mind changing it.